File: Algorithms/BOJ/Swift/1926.swift
// https://www.acmicpc.net/problem/1926
// 그림
let size = readLine()!.split(separator: " ").map {Int( String($0))! }
var grid: [[String]] = []
for _ in 0..<size[0] {
    grid.append(contentsOf: Array(arrayLiteral: readLine()!.split(separator: " ").map({ String($0) })))
}

func bfs(_ i: Int, _ j: Int) {
    if i < 0 || i >= size[0] || j < 0 || j >= size[1] || grid[i][j] != "1" {
        return
    }
    grid[i][j] = "0"
    max += 1
    
    bfs(i+1, j)
    bfs(i-1, j)
    bfs(i, j+1)
    bfs(i, j-1)
}

var cnt = 0
var max = 0
var total = 0
if grid[0].count > 0 {
    for i in 0..<size[0] {
        for j in 0..<size[1] {
            if grid[i][j] == "1" {
                max = 0
                bfs(i, j)
                cnt += 1
                if total < max {
                    total = max
                }
            }
        }
    }
}
print(cnt)
print(total)

File: Algorithms/BOJ/Python/1926.py
# https://www.acmicpc.net/problem/1926
# 그림
import sys
sys.setrecursionlimit(10000000)
n, m = map(int, sys.stdin.readline().split())
grid = []
for _ in range(n):
    grid.append(list(map(int, sys.stdin.readline().split())))

def bfs(i, j):
    global mx
    if i < 0 or i >= n or j < 0 or j >= m or grid[i][j] != 1:
        return
    
    # 현재 위치 방문
    grid[i][j] = 0
    mx += 1

    bfs(i+1, j)
    bfs(i-1, j)
    bfs(i, j+1)
    bfs(i, j-1)

cnt = 0
mx = 0
total = 0
if len(grid[0]) > 0:
    for i in range(n):
        for j in range(m):
            if grid[i][j] == 1:
                mx = 0
                bfs(i, j)
                cnt += 1
                if total < mx:
                    total = mx
print(cnt)
print(total)